server/enclave/ecalls.cpp

```diff
@@ -30,65 +31,69 @@ void copy_arr_to_enclave(unsigned char* dst[], size_t num, unsigned char* src[],
   for (int i = 0; i < num; i++) {
     size_t nlen = lengths[i];
     check_host_buffer(src[i], nlen);
-    dst[i] = (unsigned char*) strndup((const char*) src[i], nlen);
-    dst[i][nlen] = '\0';
+    dst[i] = new unsigned char[nlen];
```

Comment:
All arrays created using `new` need to be deleted later by calling `delete` to prevent a memory leak.